File: cpp/extgcd.cpp
template<typename T>
pair<T,T> extgcd(T a, T b){
    // ax + by = gcd(a, b)
    if(a == 0){
        return pair<T,T>(0, 1);
    }else if(b == 0){
        return pair<T,T>(1, 0);
    }

    T q = a/b;
    T r = a%b;

    auto [s, t] = extgcd(b, r);

    T y = s - q*t;
    return pair<T,T>(t, y);
}

File: rust/extgcd.rs
fn extgcd(a: i64, b: i64) -> (i64, i64) {
    // ax + by = gcd(a, b)
    if a == 0 { return (0, 1); }
    if b == 0 { return (1, 0); }
    let q = a/b;
    let r = a%b;
    let (s, t) = extgcd(b, r);

    let y = s - q*t;
    (t, y)
}
